pkgs/build-support/trivial-builders/default.nix

```
@@ -86,6 +86,10 @@ rec {
     , preferLocalBuild ? true
     , derivationArgs ? { }
     }:
+    assert lib.assertMsg (destination == "" || lib.hasPrefix "/" destination && destination != "/") ''
+      destination must start with a /
```

Would it make sense to name it an "absolute path" then ?

From [here](https://www.redhat.com/sysadmin/linux-path-absolute-relative):

> An absolute path always begins from the absolute start of your hard drive and describes every step you must take through the filesystem to end up at the target location.

Here's my suggestion:

```suggestion
      destination must be an absolute path, starting from the root directory (typically with a forward slash /). Ensure that the full directory structure is specified, and avoid using relative paths.
```

---

I'm not sure if that wording might not be misleading here. Destination is always relative to its store path (by just setting `target=$out${lib.escapeShellArg destination}`).

Also I'd opt for stronger wording than "avoid". This isn't a lint for best practice but rather a way to catch malformed input at eval time rather than having the derivation fail to build.

I'm open for making the message more descriptive though, perhaps something like this (still not quite happy about bending the definition of "absolute" though)

```suggestion
      destination must be an absolute path, relative to the derivation's out path. Ensure that the path starts with a / and specifies at least the filename.
```

---

Agreed ! Perfect!